### DISCUSSION TITLE
Content doesn't renders statically when using `Catch all routes` combining with lazy imports

### DISCUSSION THREAD

**Discussion by augustomallmann** [flareact/flareact#221] [Q&A]
Hello there,

I'm migrating a blog that uses Jekyll as the SSG to Flareact. I'm having difficulties handling dynamic routes when applying `catch-all routes` to keep the same url structure we have now.

For example, now we have the current structures:

```
homepage: site.com/en/blog/
post: site.com/en/blog/post-slug/
category: site.com/en/blog/category-slug/
```

The issue comes from handling the `post` and `category` pages. For what I understood reading the documentation when I have multiple routes the correct way is to create a `[...cathAll].tsx` file. So my folder structure is `/pages/[lang]/blog/[...slug].tsx`.

`category` and `post` pages use different components, so when the function `getEdgeProps` runs, I have to first check whether the route is category or post and then fetch its content.

For handling the render of the components, I thought about using the `react.lazy` import (https://reactjs.org/docs/code-splitting.html). But I notice that after implementing it, the page stopped rendering the HTML statically but started to bring only a JSON script in its source code. 

Is there a way to have this kind of dynamic routing splitting the code and keeping the content rendered server-side?

This is just a sample test I was developing as a proof of concept. If the route has `/blog`, it should render `Footer` component.

```
import React, { Suspense } from 'react';
const Footer = React.lazy(() => import('../../src/components/Footer'));

export default function Teste({ params, blog }) {
    return (
        <>
            <h1>{blog}</h1>
            <Suspense fallback={<div>Loading...</div>}>
                {blog === 'contém' ? <Footer /> : 'não'}
            </Suspense>
        </>
    );
}

export async function getEdgeProps({ params }) {
    const { teste } = params;

    let blog: string;

    teste.includes('blog') ? (blog = 'contém') : (blog = 'não contém');

    return {
        props: {
            params,
            blog
        }
    };
}
```

And the page source looks as follows:

```
<body>
    <div id="__flareact"></div>
    <script id="__FLAREACT_DATA"
        type="application/json">{"page":{"page":"./teste/[...teste].tsx","params":{"teste":["e","blog","e"]},"pagePath":"/teste/[...teste]","parts":[],"test":{}},"props":{"params":{"teste":["e","blog","e"]},"blog":"contém"}}</script>
    <script src="/_flareact/static/rlWar--YKm6lAoJHe36B-/_buildManifest.js"></script>
    <script src="/_flareact/static/main.5a58ba9e54feaa6f7de8.js"></script>
    <script src="/_flareact/static/webpack.93e75b13c12af80f38e6.js"></script>
    <script src="/_flareact/static/framework.550ea1afbc897543befd.js"></script>
    <script src="/_flareact/static/commons.52fad877657e0d8a0f64.js"></script>
    <script src="/_flareact/static/77e738bae7195dbf82ca8b55fe3b3fd035c6ec3c.4a3f3725b9d2496424a5.js"></script>
    <script src="/_flareact/static/992335e631ecb3077943b8cc4d4f368df0a72de1.67fd2896671d1a5233d2.js"></script>
    <script src="/_flareact/static/f25ea36e63b3f65f06ff7d58ab965889eae47741.9e6bad4dfb5a0e05eabb.js"></script>
    <script src="/_flareact/static/8764aea244c622a0157f9491e5f453c9d6ed3716.e832104aad58d53d7b7e.js"></script>
    <script src="/_flareact/static/25e991e83468c1b90859dc595a531164a7295902.b6c338c6cf122857d850.js"></script>
    <script src="/_flareact/static/pages/_app.af71bf76c26761059994.js"></script>
    <script src="/_flareact/static/pages/teste/[...teste].4020e51eb2da71b9fc6d.js"></script>
</body>
```

**Comment by traffisco** [comment]
Hey,
Dynamic component loading is not supported.
You either can route to a different file(eg `/blog/post/[slug].js` and `/blog/category/[slug].js`) or load all components and render only the ones needed for a specific route.

**Reply by robsongajunior** [reply]
ok, understood!
I have another question...
Is it forever behavior ou just now will not possible?

**Reply by traffisco** [reply]
Code is compiled and uploaded to cloudflare to live there forever until updated. Dynamic imports are impossible.
That doesn’t necessarily means that some library can’t mimic the desired functionality for ease of development.
Does it exist? No clue

**Comment by traffisco** [comment]
I think we can actually mimic the desired functionality.
We can't really do dynamic component loading but we can "not load" the bundles that are not needed.
We can do a separate bundling for the imports that suppose to be dynamic at build time and load/not load the scripts in the browser on demand. It will achieve the same goal(browser bundle size).
Any thoughts @tj-kev?

**Reply by tj-kev** [reply]
Sounds like a good idea to me. How much work do you think that would be to modify the webpack configs?

Interested to hear @jplhomer views?

**Reply by jplhomer** [reply]
Hey! Not sure I entirely understand the goal, but it sounds like a good use case for React 18 streaming SSR (powered by Suspense on the server) which is in RC, and React Server Components, which is still experimental. 

My gut says it's going to be a lot of complex wiring to mimic the intended behavior. But definitely open to seeing what the solution looks like.
